senders/solomon/solomon.go

```diff
@@ -226,38 +240,123 @@ func (s *solomonClient) Send(task tasks.DataType, timestamp uint64) error {
 		}
 		logger.Debugf("%s Data to POST: %s", s.id, string(push))
 
-		req, err := http.NewRequest("POST", s.api, bytes.NewBuffer(push))
+		logger.Debugf("%s Sending work to JobQueue", s.id)
+		JobQueue <- Job{PushData: push, SolCli: s}
+
+	}
+
+	return nil
+}
+
+func NewSolomonClient(cfg *SolomonCfg, id string) (ss SolomonSender, err error) {
+	ss = &solomonClient{
+		id:                 id,
+		api:                cfg.Api,
+		cluster:            cfg.Cluster,
+		project:            cfg.Project,
+		fields:             cfg.Fields,
+		connection_timeout: cfg.Connection_timeout,
+		rw_timeout:         cfg.Rw_timeout,
+	}
+
+	return
+}
+
+func (w Worker) Start(d *Dispatcher) {
+	for {
+		// add current worker to the WorkerPool on first iteration
+		// after finishing select statement worker will add
+		// itself back to the pool again
+		d.WorkerPool <- w.JobChannel
+		select {
+		// wait for a job in workers JobChannel
+		case job := <-w.JobChannel:
```

Comment:
I don't see a way to terminate Start(), so it can be more clear to have something like
```go
for job := range w.JobChannel
```